packages/@aws-cdk/aws-apigatewayv2/lib/websocket/stage.ts

```diff
@@ -0,0 +1,67 @@
+import { Resource, Stack } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnStage } from '../apigatewayv2.generated';
+import { IStage } from '../common';
+import { IWebSocketApi } from './api';
+
+const DEFAULT_STAGE_NAME = 'dev';
+
+/**
+ * Represents the WebSocketStage
+ */
+export interface IWebSocketStage extends IStage {
+}
+
+/**
+ * Properties to initialize an instance of `WebSocketStage`.
+ */
+export interface WebSocketStageProps {
+  /**
+   * The WebSocket API to which this stage is associated.
+   */
+  readonly webSocketApi: IWebSocketApi;
+
+  /**
+   * The name of the stage.
+   */
+  readonly stageName: string;
+
+  /**
+   * Whether updates to an API automatically trigger a new deployment.
+   * @default false
+   */
+  readonly autoDeploy?: boolean;
+}
+
+/**
+ * Represents a stage where an instance of the API is deployed.
+ * @resource AWS::ApiGatewayV2::Stage
+ */
+export class WebSocketStage extends Resource implements IWebSocketStage {
```

Comment:
Looks so much like our existing http api implementation - https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-apigatewayv2/lib/http/stage.ts. Even the part of API mapping is going to look similar.

I'm mulling over how much of this we should re-use across the two, rather than duplicate. WDYT?